### DISCUSSION TITLE
How to set environment variables for host-containers?

### DISCUSSION THREAD

**Discussion by dpavlov-smartling** [bottlerocket-os/bottlerocket#4315] [Q&A] [answered]
Hello,

After checking the documentation, github issues and discussions I didn't find any native way to set environment variables for host-containers.
As of now, it seems that the only way is to specify the needed values in the host-container's user-data and then propagate them in the container's entrypoint.
My use case is monitoring of the instance including its metrics and logs:
1. To get Bottlerocket logs from journald I use custom Splunk container
2. To collect rest of the metrics I want to use Datadog

The reason for runnning them as a host-containers is to make sure that we have logs and metrics even if the instance hasn't joined the cluster.
This is for audit purpose.

The ability to add environment variables for host-containers would eliminate the need to create custom containers that can process user-data. Instead you would be able to use original container and configure it via environment variables. 

Thanks,

**Comment by vigh-m** [comment, marked as answer]
Thanks for reaching out. Setting environment variables on host-containers isn't currently supported in host-containers on Bottlerocket. It is a good feature that we will consider implementing in the future. For now, I have created [an issue](https://github.com/bottlerocket-os/bottlerocket/issues/4329) on our end that we can track and reference for this.

**Comment by dpavlov-smartling** [comment]
Hello @vigh-m ,
Thank you for the update and created issue. I also want to add that it looks like it is impossible to limit host-container resources like CPU or RAM. Nobody wants to be affected if host-container like Splunk got some memory leak and eats all the RAM on the instance.
Do you think request to add such an options should be added to the same issue that you have created or it should be the new one?

Thanks
